boards/common/blxxxpill/Makefile.include

```diff
@@ -8,20 +8,15 @@ ifeq (dfu-util,$(PROGRAMMER))
 
   # Leave some space at the beginning of the flash for the bootloader
   ROM_OFFSET ?= 0x2000
-
+  FFLAGS_OPTS ?= --dfuse-address $(FLASH_ADDR):leave
```

Comment:
I see a potential issue with moving this as is at board level: `FLASH_ADDR` is computed in `dfu-util.inc.mk`, so after the board configuration is included. It works now because of make deferred evaluation but this is risky.

I think I would prefer to introduce a dedicated variable: `DFU_USE_DFUSE` that is equal to 1 by default (since most of our boards are using it) and add `--dfuse-address $(FLASH_ADDR):leave` to `FFLAGS_OPTS` only if `DFU_USE_DFUSE` is equal to 1. This logic should be placed in `dfu-util.inc.mk`.

Boards that are not using `DfuSe` could then just set `DFU_USE_DFUSE ?= 0` in their `Makefile.include`.

This is the cleanest and most flexible approach IMHO.

Reply:
I don't really like the idea, tbh. I don't find it 'clean' but rather confusing. Can't we find a way to ensure board configuration is included before dfu-util.inc.mk is parsed instead ?

Reply:
> Can't we find a way to ensure board configuration is included before dfu-util.inc.mk is parsed instead ?

It's already done like this. But the problem is that `FLASH_ADDR` is set in `dfu-util.inc.mk`, so potentially not known when you use it in the board `Makefile.include`.
Having a dedicated variable to specify that DfuSe is used by the board is not confusing it's explicit.

Reply:
Then, how about moving $(FLASH_ADDR) resolution at board level instead ? This way we can let the --dfuse-adress param at board level too.

We can even be more strict, and completely remove this resolution to hardcode the value. This value is not expected to change as those boards come with preflashed bootloader. The start addr for user application will always remain the same and this will have no impact to the code generation. Less shell call from makefile is a good thing, right ? :)

What do you think ?

Reply:
> how about moving $(FLASH_ADDR) resolution at board level instead ?

Because it's computed from ROM/RAM information that is computed in CPU Makefile.include. And this is done after boards.
You should really do what I suggested :)

Reply:
Made the change.

Reply:
Thanks!